### DISCUSSION TITLE
OpenUSD and Hydra renderer idea

### DISCUSSION THREAD

**Discussion by WangHoi** [DiligentGraphics/DiligentEngine#316] [General]
Will it be useful to provide documentation about workflow from DCC to Diligent Engine.

For example, how to export from blender?
https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon

If Diligent Engine can live reload editing scene from Blender, it will be super cool.

**Comment by TheMostDiligent** [comment]
Do you mean using Hydrogent render delegate in Blender or do you mean synchronizing USD scene edited in blender in USD viewer?
Both ways should be possible.

**Comment by WangHoi** [comment]
Blender already has "Hydra Storm" rasterizer and "Hydra AMD ProRender" render delegate.
Add the Hydrogent renderer addon to blender seems not interesting.

Export from Blender to Diligent Engine is quite challenging.
- Export mesh and convert basic shader nodes to MaterialX material is not much work.
- The hard part is exporting lightmaps, reflection probes and irradiance volumes from Blender and regenerating the indirect lighting in Diligent Engine

I suspect it's a matter of trial and error, there are no shortcuts and magic ?

**Reply by TheMostDiligent** [reply]
Yes, exporting lighting setup might be challenging. USD provides a number of standard light source types.
Hydrogent currently uses screen-space approximations for GI (reflections, ambient occlusion). Exporting refelction probes and irradiance volumes is not implemented now.
